tests/test_util.py

```diff
@@ -494,3 +498,35 @@ def test_running_test_fixture():
     Assert that the RUNNING_TESTS variable is set to True when we run the tests
     """
     assert inmanta.RUNNING_TESTS
+
+
+async def test_threadpool_join():
+    tp = ThreadPoolExecutor()
+
+    hanglock = Event()
+    done = Queue()
+
+    eventloop = asyncio.get_event_loop()
+
+    async def cor():
+        await asyncio.sleep(0.02)
+
+    def worker():
+        # hang on lock
+        hanglock.wait()
+        # hang on ioloop
+        block = asyncio.run_coroutine_threadsafe(cor(), loop=eventloop)
+        block.result()
+        done.put("A")
+
+    for i in range(5):
+        tp.submit(worker)
+
+    tp.shutdown(wait=False)
+    assert done.qsize() == 0
+    hanglock.set()
+    assert done.qsize() == 0
```

Comment:
I think I don't understand how this assertion can always be true. I played around a bit with the testcase and set the sleep in `cor()` to 0 and the assertion is still always true.

Reply:
When unblocking the hanglock, all thread get stuck on `block.result()` 
for this to unblock, the co-routine `cor` has to run. 

Which means that the ioloop must be able to execute things. 
But we are sitting on the main thread, so it can't execute until we do an `await` to return control to the ioloop....

Reply:
I see, thanks :+1: